activation/handler_v1.go

```diff
@@ -576,7 +560,8 @@ func (h *HandlerV1) processATX(
 		return fmt.Errorf("%w: validating atx %s (deps): %w", pubsub.ErrValidationReject, watx.ID(), err)
 	}
 
-	if err := h.storeAtx(ctx, atx, watx); err != nil {
+	publishing := h.local == peer
+	if err := h.storeAtx(ctx, atx, watx, publishing); err != nil {
```

Comment:
Passing a boolean makes it more difficult to understand what is going on in tests (its not immediately obvious what `true` and `false` mean). How about we pass `peer` into `storeAtx` and `checkMalicious` so in tests we can either pass `atxHdlr.local` or `peer.ID("other")` to make clear if the ATX is published by self or not?

Reply:
Changed to passing peer ID